### PR TITLE
GeoLocationFilter: fix autocomplete

### PR DESCRIPTION
**Pull request by oshi97**
This commit fixes the autocomplete for the geolocation filter
by replacing barKey (deprecated) with searchParameters when
querying filtersearch. It also replaces all other usages of
barKey, and fixes a few typos in readme/css class selector.

SearchParameters was also added to the readme for the geolocation
filter.

TEST=manual
Ran dev site and tested that autocomplete was functioning properly.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -782,9 +782,18 @@ Displays a "Use My Location" button that filters results to a radius around the
 ```js
 ANSWERS.addComponent('', {
```
Comment (tmeyer2115) [inline README.md]: Make this 'GeoLocationFilter' instead of ''

File: README.md [hunk 2]
```diff
@@ -782,9 +782,18 @@ Displays a "Use My Location" button that filters results to a radius around the
...
   container: '.geolocation-filter-container',
+  // Search parameters for the geolocation autocomplete
+  searchParameters: {
```
Comment (tmeyer2115) [inline README.md]: Should the 'sectioned' attribute be added here too?

File: src/core/core.js [hunk 3]
```diff
@@ -219,11 +219,10 @@ export default class Core {
    * @param {string} input       the string to autocomplete
    * @param {string} namespace the namespace to use for the storage key
    * @param {string} verticalKey the vertical key for the experience
```
Comment (tmeyer2115) [inline src/core/core.js:221]: Do we need this to now take in searchParameters?

File: src/core/search/autocompleteapi.js [hunk 4]
```diff
@@ -70,22 +70,21 @@ export default class AutoCompleteApi {
         'experienceKey': this._experienceKey,
         'version': this._experienceVersion,
         'verticalKey': config.verticalKey,
-        'inputKey': config.barKey,
```
Comment (tmeyer2115) [inline src/core/search/autocompleteapi.js:73]: Is inputKey deprecated?

File: src/core/search/autocompleteapi.js [hunk 5]
```diff
@@ -70,22 +70,21 @@ export default class AutoCompleteApi {
...
       .then(response => response.json())
-      .then(response => AutoCompleteDataTransformer.filter(response.response, config.barKey))
+      .then(response => AutoCompleteDataTransformer.filter(response.response))
       .catch(error => {
         throw new AnswersEndpointError('Filter search request failed', 'AutoComplete', error);
       });
   }
 
   /** @inheritdoc */
-  queryVertical (input, verticalKey, barKey) {
```
Comment (tmeyer2115) [inline src/core/search/autocompleteapi.js:88]: Does this now need to take in searchParameters?

File: src/core/core.js (hunk 3 above)
Comment (oshi97) [reply src/core/core.js:221]: ah I'm a dingus

File: src/core/search/autocompleteapi.js (hunk 4 above)
Comment (oshi97) [reply src/core/search/autocompleteapi.js:73]: Yup asked jesse!

File: src/core/search/autocompleteapi.js (hunk 5 above)
Comment (oshi97) [reply src/core/search/autocompleteapi.js:88]: Yes

**Review by aginsberg — DISMISSED** [review]
LGTM! 🌐

**Review by tmeyer2115 — DISMISSED** [review]

**Review by tmeyer2115 — APPROVED** [review]

**Review by data-enabler — COMMENTED** [review]
lgtm, though I still have no idea what barKey was supposed to do.

**Comment by jsharps-yext** [comment]
It dates back to Think V1 and identifies a search bar on the page. For instance, you might have a page that was [ Search for Doctors] [Search for Locations] [Submit] and the two search bars would have different filters and thus you'd need to identify them to the backend.

Now if you want multiple search bars you usually don't. We migrated away from that. And we decided for the one case that still needs more information, filterSearch, the frontend would define it instead of the backend.

**Comment by jsharps-yext** [comment]
Not to be confused with the thing that lets you back in after last call.

**Pull request merged**